torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -18,6 +18,19 @@ c10::intrusive_ptr<ProcessGroup::Work> broadcast(
           root_rank, root_tensor, std::chrono::milliseconds(timeout)});
 }
 
+c10::intrusive_ptr<ProcessGroup::Work> allreduce_(
```

Comment:
just a heads-up that there is an ongoing effort on consolidating `ProcessGroup::Work` with torch Future. 

cc @kumpera

Reply:
In the last conversation, we agree that we won't have `wait()` in the graph. What we do is to fulfill the async handler immediately with the tracing proxy.